Tests/JWTKitTests/PSSTests.swift

```diff
@@ -20,7 +20,7 @@ final class PSSTests: XCTestCase {
             exp: .init(value: .init(timeIntervalSince1970: 2_000_000_000))
         )
 
-        let token = try await keyCollection.sign(payload, kid: "private")
+        let token = try await keyCollection.sign(payload, with: .init(fields: ["kid": .string("private")]))
```

Comment:
My gut tells me we should try and tighten up this API for the common use case, either via syntactic sugar or helper functions that retain the old API and ferry it down to the new API